gunicorn/arbiter.py

```diff
@@ -186,10 +187,11 @@ def init_signals(self):
         # initialize all signals
         for s in self.SIGNALS:
             signal.signal(s, self.signal)
-        signal.signal(signal.SIGCHLD, self.handle_chld)
 
     def signal(self, sig, frame):
-        if len(self.SIG_QUEUE) < 5:
+        if sig is signal.SIGCHLD:
+            self.SIG_CHLD = True
```

Comment:
One consequence of not using the signal queue, as compared to the old implementation, is that we're not reaping immediately. We could call `self.wakeup()` here.

Alternatively, is there any reason for the limit, below, to 5 signals in the queue? If not, we could have no special case for `SIGCHLD`.